core/base-shaper.lua

```diff
@@ -29,6 +29,7 @@ SILE.shapers.base = std.object {
       SILE.settings.temporarily(function()
         SILE.settings.set("font.size", options.size)
         SILE.settings.set("font.family", options.family)
+        if options.filename then SILE.settings.set("font.filename", options.filename) end
```

Comment:
Why does this like need to be wrapped in an `if` statement and the preceding ones do not?

Reply:
Short answer: Because it crashed without it.

Long answer: I believe it has to do with the fact, that when the font is specified by filename here, the font family is specified by empty string. But when it is specified by the family, the filename is `nil`, which crashes the `set` function.

As to why that is, I don't know. I don't have the time right now to understand this part of the SILE, so I stopped working on this when I got this fix/workaround.

Reply:
I wonder if #607 will fix the issue with `set()`.

Reply:
I have verified both that your fix is needed and that the changes in the earlier PR fix the settings so that a nil value does not cause a crash. This can be simply:

```suggestion
        SILE.settings.set("font.filename", options.filename)
```